docs/pages/contributing/documentation/how-to-contribute.mdx

```diff
@@ -1,168 +1,367 @@
+
+### Internal contributors
+
+If you work at Teleport:
+
+1. Add files and commit changes to your local branch periodically with commands similar to the following:
+
+    ```code
+    $ git add <new-file-name>                      # Add a specific new file to be committed
+    $ git add -A                                   # Add all changed files to the list of files to be committed
+    $ git commit -am "Fix or feature description"  # Add and commit changes with a comment
+    $ git commit -m                                # Commit changes already added
+    ```
+
+1. Push your branch to `gravitational/teleport` with a command similar to the following:
+    
+    ```code
+    $ git push origin my-branch
+    ```
+
+1. Open the [Teleport repository](https://github.com/gravitational/teleport), select your branch as the branch to 
+merge into main, then click **New pull request**.
+
+### External contributors
+
+If you are an external contributor, you must commit your changes to your branch and push your branch to a 
+fork of the `gravitational/teleport` repository.
+
+To push changes as an external contributor:
+
+1. Verify that you have an SSH key pair and have stored your public key in GitHub.
+
+1. Open the [Teleport repository](https://github.com/gravitational/teleport), click **Fork**, then select **Create a new fork**.
+
+1. Verify the owner and repository name.
+
+1. Click **Create fork**.
+    
+    Alternatively, you can create a fork from the command-line by running a command similar to the following:
+    
+    ```code
+    $ git remote add fork ssh://git@github.com/my-user/teleport
+    ```
+
+    In this command, `my-user` represents your GitHub user name and `teleport` is the name of your Teleport repository fork.
+
+1. Pull all of the changes from the remote repository into your local fork by running the following command:
+    
+    ```code
+    $ git fetch origin
+    ```
+
+1. Push changes from your local branch to the remote fork of the [Teleport repository](https://github.com/gravitational/teleport) 
+by running a command similar to the following:
+    
+    ```code
+    $ git push --set-upstream fork my-branch
+    ```
+
+1. Open the [Teleport repository](https://github.com/gravitational/teleport), select your branch as the branch to merge into main, 
+then click **New pull request**.
+
+    The continuous integration (CI) pipeline automatically assign reviewers to your pull request. 
+    To ensure that your PR is merged, you should respond to reviewer feedback in a timely manner. 
+    If you don't respond to reviewer feedback in a reasonable period of time, your pull request is likely to 
+    be deemed inactive and closed automatically. 
+
+1. Wait for the minimum required approvals, then merge your pull request as soon as possible.
+
```

Comment:
For external contributors, it might be worth noting that you should expect a "Buddy PR", where we open an internal branch that includes your commits. This is because a lot of our CI jobs only run for internal contributors